### PR TITLE
Move show media scan notifications to AppPreferences interface

### PR DESCRIPTION
**Pull request by ezaquarii**

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@ezaquarii looks good to me code-wise, can you sign the commit to make the do bot happy? 😃

**Comment by ezaquarii** [comment]
I fixed the sign-offs for all pending PRs.

File: src/main/java/com/owncloud/android/ui/activity/Preferences.java [hunk 1]
```diff
@@ -101,28 +102,25 @@
...
-    private String mStoragePath;
+    private ListPreference prefStoragePath;
+    private String storagePath;
     private String pendingLock;
 
-    private Account mAccount;
-    private ArbitraryDataProvider mArbitraryDataProvider;
+    private Account account;
+    private ArbitraryDataProvider arbitraryDataProvider;
+    private AppPreferences preferences;
```
[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/activity/Preferences.java]: Hmm, this is somewhat true. I'd say that to have a proper Naming schema than an Activity should be suffix with `Activity` so the class should be renamed, while I'd be fine with postponing that and having it done in a separate PR (I'd also be fine with doing it here, whatever you prefer)

What do you think @ezaquarii @tobiasKaminsky ?

**Comment by AndyScherzinger** [comment]
@ezaquarii can you rebase/resolve-conflicts and @ezaquarii @tobiasKaminsky can we agree on the way forward regarding https://github.com/nextcloud/android/pull/3697#discussion_r262401023 ?

File: src/main/java/com/owncloud/android/ui/activity/Preferences.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/activity/Preferences.java]: Having Fragment/Activity/Adapater/… as suffix for class is in my opinion a good way.
If we do a change to Preferences.java I would suggest to name it SettingsActivity.java as this then reflects the entry in drawer.

[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/activity/Preferences.java]: Sounds good to me. Do that change in a separate PR or wihtin this PR to get rid of the codacy warning?

[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/activity/Preferences.java]: I would do this separate and then merge this PR after the other.

[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/activity/Preferences.java]: https://github.com/nextcloud/android/pull/3705

[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/ui/activity/Preferences.java]: #3705 has been merged

**Comment by AndyScherzinger** [comment]
🎉 Next one merged 🚀

**Pull request merged**